### PR TITLE
Add local ID settings struct

### PR DESCRIPTION
**Pull request by gordonwang0**

### CODE REVIEW

File: identity/aziot-identityd/src/error.rs [hunk 1]
```diff
@@ -19,84 +19,87 @@ impl Error {
...
-    ParseSettings(toml::de::Error),
-    SaveDeviceInfo(std::io::Error),
-    SaveSettings(std::io::Error),
+	CreateCertificate(Box<dyn std::error::Error + Send + Sync>),
+	CreateHomeDir(std::io::Error),
+	InvalidUri(http::uri::InvalidUri),
+	LoadKeyOpenslEngine(openssl2::Error),
+	LoadDeviceInfo(std::io::Error),
+	LoadSettings(std::io::Error),
+	BadSettings(std::io::Error),
```
Comment (gordonwang0) [inline identity/aziot-identityd/src/error.rs:60]: Addition of BadSettings is the only non-spacing change in this file.

Comment (CindyXing) [reply identity/aziot-identityd/src/error.rs:60]: Ok. Are we using rustfmt when submitting PR?

**Review by CindyXing — APPROVED** [review]

File: identity/aziot-identityd/test/good_auth_settings.toml [hunk 2]
```diff
@@ -4,6 +4,10 @@
 hostname = "iotedge"
 homedir = "/var/lib/aziot/identityd"
 
+# Local ID settings
```
Comment (massand) [inline identity/aziot-identityd/test/good_auth_settings.toml:7]: Please add a comment indicating that this is concatenated (suffixed) to id.name for producing the CN in the local identity's X.509 certificate.

Comment (gordonwang0) [reply identity/aziot-identityd/test/good_auth_settings.toml:7]: Done

**Review by massand — APPROVED** [review]

**Pull request merged**